### PR TITLE
PS-3773: terminate jobs in project command

### PR DESCRIPTION
**Pull request by MiroCillik**
https://keboola.atlassian.net/browse/PS-3773

A command to terminate all jobs in a project in specified state. For example to terminate all "waiting" jobs in a project in case of an incident.

Changes:

- feat: Terminate queue jobs in project in specified state

---

Additional notes

:warning:  *Don't forget to release new version after merge*

### CODE REVIEW

**Comment by MiroCillik** [comment]
Note to self: add readme

File: src/Keboola/Console/Command/QueueMassTerminateJobs.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            (new ListJobsOptions())
+                ->setStatuses(['waiting'])
+                ->setLimit(3000)
+        );
+
+        $terminatedJobsIds = [];
+        foreach ($jobs as $job) {
+            try {
+                $jobQueueClient->terminateJob($job['id']);
+                $terminatedJobsIds[] = $job['id'];
```
Comment (pivnicek) [inline src/Keboola/Console/Command/QueueMassTerminateJobs.php:66]: It all looks good to me, but do you want to maybe add a log line `Terminated job $job['id']`?

**Review by pivnicek — APPROVED** [review]

File: src/Keboola/Console/Command/QueueMassTerminateJobs.php (hunk 1 above)
Comment (MiroCillik) [reply src/Keboola/Console/Command/QueueMassTerminateJobs.php:66]: OK

File: src/Keboola/Console/Command/QueueMassTerminateJobs.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+use Keboola\JobQueueClient\ListJobsOptions;
+use Keboola\StorageApi\Client as StorageClient;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class QueueMassTerminateJobs extends Command
+{
+    const ARGUMENT_STORAGE_TOKEN = 'storage-token';
```
Comment (joker806) [inline src/Keboola/Console/Command/QueueMassTerminateJobs.php:17]: Nazev commandu by mel byt uvedeny v `$defaultName` property, aby se neinicializoval pri pousteni jinych commandu (https://symfony.com/doc/5.4/console/commands_as_services.html)

Comment (joker806) [reply src/Keboola/Console/Command/QueueMassTerminateJobs.php:17]: (v `configure` potom uz neni potreba)

File: src/Keboola/Console/Command/QueueMassTerminateJobs.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    const ARGUMENT_CONNECTION_URL = 'connection-url';
+    const ARGUMENT_JOB_STATUS = 'job-status';
+
+    protected function configure()
+    {
+        $this
+            ->setName('queue:terminate-jobs')
```
Comment (joker806) [inline src/Keboola/Console/Command/QueueMassTerminateJobs.php:24]: Mozna `queue:terminate-job-projects`, kdyz to muze killnout jen joby z jednoho projektu?

File: src/Keboola/Console/Command/QueueMassTerminateJobs.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        $queueApiUrl = str_replace('connection', 'queue', $kbcUrl);
+
+        $jobQueueClient = new JobQueueClient(
+            $queueApiUrl,
+            $storageToken
+        );
+
+        $jobs = $jobQueueClient->listJobs(
+            (new ListJobsOptions())
+                ->setStatuses(['waiting'])
```
Comment (joker806) [inline src/Keboola/Console/Command/QueueMassTerminateJobs.php:58]: Tady by asi mel byt `$status` misto `['waiting']`? :slightly_smiling_face:

File: src/Keboola/Console/Command/QueueMassTerminateJobs.php (hunk 1 above)
Comment (joker806) [reply src/Keboola/Console/Command/QueueMassTerminateJobs.php:66]: Definitely! Command should provide contious output, telling it's progress and some ala success message at the end so you know it was finished correctly.

File: src/Keboola/Console/Command/QueueMassTerminateJobs.php (hunk 3 above)
Comment (pivnicek) [reply src/Keboola/Console/Command/QueueMassTerminateJobs.php:25]: `terminate-project-jobs`

Comment (joker806) [reply src/Keboola/Console/Command/QueueMassTerminateJobs.php:25]: LOL :laughing: jasne

**Review by odinuv — APPROVED** [review]
jop, funguje to

File: src/Keboola/Console/Command/QueueMassTerminateJobs.php (hunk 4 above)
Comment (MiroCillik) [reply src/Keboola/Console/Command/QueueMassTerminateJobs.php]: JJ, to uz jsem mezitim fixnul :))

File: src/Keboola/Console/Command/QueueMassTerminateJobs.php (hunk 1 above)
Comment (MiroCillik) [reply src/Keboola/Console/Command/QueueMassTerminateJobs.php]: Done

**Pull request merged**